src/cli/commands/install.js

```diff
@@ -572,14 +572,44 @@ export class Install {
   }
 
   /**
+   * Remove offline tarballs that are no longer required
+   */
+
+  async pruneOfflineMirror(lockfile: Object): Promise<void> {
+    const mirror = this.config.getOfflineMirrorPath();
+    if (!mirror) {
+      return;
+    }
+
+    const requiredTarballs = new Set();
+    for (const dependency in lockfile) {
+      const resolved = lockfile[dependency].resolved;
+      if (resolved) {
+        requiredTarballs.add(resolved.split('#')[0]);
```

Comment:
The tests would fail if there are breaking changes.
I suppose it should be fine.